### ISSUE TITLE
Anomaly "Uncaught exception Not_found." in custom notation entry

### ISSUE THREAD

**Issue by anton-trunov** [coq/coq#13966]
#### Description of the problem

The following snippet causes an anomaly:

```coq
Inductive expr  := Const : nat -> expr.

Declare Custom Entry expr.
Notation "'[' e ']'" := e (e custom expr at level 0).
Notation "x" := (Const x) (in custom expr at level 0, x bigint).
Notation "x" := x (in custom expr at level 0, x ident).
```

Here is the output:
```
Error: Anomaly "Uncaught exception Not_found."
Please report at http://coq.inria.fr/bugs/.
```

I'm wondering if it's possible to parse embedded numerals and identifiers without adding additional syntax to distinguish between those?

#### Coq Version

8.13.1

**Comment by ejgallego** [comment]
```ocaml
Error: Anomaly "Uncaught exception Not_found."
Please report at http://coq.inria.fr/bugs/.
Raised at CList.remove_first in file "clib/cList.ml", line 660, characters 10-25
Called from CList.remove_first in file "clib/cList.ml", line 659, characters 19-35
Called from Notation.keymap_remove in file "interp/notation.ml", line 370, characters 17-77
Called from Notation.remove_uninterpretation in file "interp/notation.ml", line 1451, characters 25-100
Called from Notation.declare_notation in file "interp/notation.ml", line 1748, characters 16-92
Called from Metasyntax.open_notation in file "vernac/metasyntax.ml", line 1403, characters 18-123
Called from Lib.add_anonymous_leaf in file "library/lib.ml", line 250, characters 4-28
Called from Metasyntax.add_notation_in_scope in file "vernac/metasyntax.ml", line 1603, characters 2-46
Called from Metasyntax.add_notation in file "vernac/metasyntax.ml", line 1702, characters 4-66
Called from Vernacinterp.interp_typed_vernac in file "vernac/vernacinterp.ml", line 28, characters 19-23
Called from Vernacinterp.interp_control.(fun) in file "vernac/vernacinterp.ml", line 215, characters 24-69
Called from Flags.with_modified_ref in file "lib/flags.ml", line 17, characters 14-17
Re-raised at Exninfo.iraise in file "clib/exninfo.ml", line 82, characters 4-38
Called from Vernacinterp.interp_gen.(fun) in file "vernac/vernacinterp.ml", line 264, characters 18-43
Called from Vernacinterp.interp_gen in file "vernac/vernacinterp.ml", line 262, characters 6-279
Re-raised at Exninfo.iraise in file "clib/exninfo.ml", line 82, characters 4-38
Called from Stm.Reach.known_state.reach.(fun) in file "stm/stm.ml", line 2154, characters 16-43
Called from Stm.State.define in file "stm/stm.ml", line 957, characters 6-10
Re-raised at Exninfo.iraise in file "clib/exninfo.ml", line 82, characters 4-38
Called from Stm.Reach.known_state.reach in file "stm/stm.ml", line 2296, characters 4-105
Called from Stm.observe in file "stm/stm.ml", line 2397, characters 4-60
Re-raised at Exninfo.iraise in file "clib/exninfo.ml", line 82, characters 4-38
Called from Stm.finish in file "stm/stm.ml", line 2408, characters 12-50
Called from Stm.process_transaction.(fun) in file "stm/stm.ml", line 2669, characters 22-45
Re-raised at Exninfo.iraise in file "clib/exninfo.ml", line 82, characters 4-38
Called from Stm.add in file "stm/stm.ml", line 2766, characters 8-50
Called from Vernac.interp_vernac in file "toplevel/vernac.ml", line 61, characters 28-90
Re-raised at Exninfo.iraise in file "clib/exninfo.ml", line 82, characters 4-38
Called from Coqloop.process_toplevel_command in file "toplevel/coqloop.ml", line 400, characters 17-62
Called from Coqloop.read_and_execute_base in file "toplevel/coqloop.ml", line 437, characters 4-39
Called from Coqloop.read_and_execute in file "toplevel/coqloop.ml", line 443, characters 6-34
```

**Comment by herbelin** [comment]
Sorry for the regression and thanks for reporting. This is fixed in #14377.

**Comment by anton-trunov** [comment]
Thank you, Hugo!
